src/SFA.DAS.AANHub.Application/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandler.cs

```diff
@@ -0,0 +1,120 @@
+﻿using MediatR;
+using SFA.DAS.AANHub.Application.Common;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+using SFA.DAS.AANHub.Domain.Common;
+using SFA.DAS.AANHub.Domain.Entities;
+using SFA.DAS.AANHub.Domain.Interfaces;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+using System.Text.Json;
+
+namespace SFA.DAS.AANHub.Application.CalendarEvents.Commands.PutCalendarEvent;
+public class PutCalendarEventCommandHandler : IRequestHandler<PutCalendarEventCommand, ValidatedResponse<SuccessCommandResult>>
+{
+    private readonly ICalendarEventsWriteRepository _calendarEventWriteRepository;
+    private readonly IAuditWriteRepository _auditWriteRepository;
+    private readonly IAttendancesWriteRepository _attendancesWriteRepository;
+    private readonly INotificationsWriteRepository _notificationsWriteRepository;
+    private readonly IMembersReadRepository _membersReadRepository;
+    private readonly IAanDataContext _aanDataContext;
+
+    public PutCalendarEventCommandHandler(
+        ICalendarEventsWriteRepository calendarEventWriteRepository,
+        IAuditWriteRepository auditWriteRepository,
+        IAttendancesWriteRepository attendancesWriteRepository,
+        INotificationsWriteRepository notificationsWriteRepository,
+        IMembersReadRepository membersReadRepository,
+        IAanDataContext aanDataContext)
+    {
+        _calendarEventWriteRepository = calendarEventWriteRepository;
+        _auditWriteRepository = auditWriteRepository;
+        _attendancesWriteRepository = attendancesWriteRepository;
+        _notificationsWriteRepository = notificationsWriteRepository;
+        _membersReadRepository = membersReadRepository;
+        _aanDataContext = aanDataContext;
+    }
+
+    public async Task<ValidatedResponse<SuccessCommandResult>> Handle(PutCalendarEventCommand command, CancellationToken cancellationToken)
+    {
+        var existingEvent = await _calendarEventWriteRepository.GetCalendarEvent(command.CalendarEventId);
+
+        var audit = new Audit()
+        {
+            Action = "Put",
+            Before = JsonSerializer.Serialize(existingEvent),
+            ActionedBy = command.AdminMemberId,
+            AuditTime = DateTime.UtcNow,
+            Resource = nameof(CalendarEvent),
+        };
+
+        UpdateCalendarEventToNewValues(command, existingEvent);
+
+        audit.After = JsonSerializer.Serialize(existingEvent);
+        _auditWriteRepository.Create(audit);
+
+        if (command.SendUpdateEventNotification)
+        {
+            var existingAttendances =
```

Comment:
We are not updating attendance so this should use read repository

Reply:
nice spot!

Reply:
I have addressed this, but for some reason GitHub is not marking it as outdated, so leaving it 'unresolved' so you can check.